src/FortuneWindow.cpp

```diff
@@ -12,49 +15,40 @@
 #define M_ABOUT_REQUESTED 'abrq'
 
 FortuneWindow::FortuneWindow(void)
-	: BWindow(BRect(0,0,300,300),"Fortune",B_DOCUMENT_WINDOW,B_ASYNCHRONOUS_CONTROLS),
+	: BWindow(BRect(0,0,300,300), "Fortune",
+		B_DOCUMENT_WINDOW, B_NOT_RESIZABLE | B_NOT_ZOOMABLE |
+		B_NOT_MINIMIZABLE | B_AUTO_UPDATE_SIZE_LIMITS),
 	fFortune()
 {
-	BPath path;	
+	BPath path;
 	find_directory(B_SYSTEM_DATA_DIRECTORY, &path);
 	path.Append("fortunes");
 
 	fFortune.SetFolder(path.Path());
 
 	BView *back = new BView(Bounds(),"background",B_FOLLOW_ALL, B_WILL_DRAW);
```

Comment:
If you use layout management, all BViews, BButtons, BTextViews etc. need to use the layout-aware versions, i.e. without the BRect as 1st parameter. See the HaikuBook at https://www.haiku-os.org/docs/api/